packages/react-router-dom/server.tsx

```diff
@@ -348,15 +348,19 @@ function createHref(to: To) {
 }
 
 function encodeLocation(to: To): Path {
-  // Locations should already be encoded on the server, so just return as-is
```

Comment:
This was a bad assumption - while they're properly encoded in `request.url`, we decode paths for matching and this function is used to re-encode in external-facing paths.